### PR TITLE
Feature/106 create project page

### PR DESCRIPTION
**Pull request by todd-dembrey**

### CODE REVIEW

File: opentech/public/utils/models.py [hunk 1]
```diff
@@ -259,3 +260,39 @@ class Meta:
...
+        ]),
+        PageChooserPanel('source', ['public_funds.FundPage', 'public_funds.LabPage']),
+    ]
+
+    class Meta(Orderable.Meta):
+        abstract = True
+
+
+class FundingMixin(models.Model):
+    '''Impliments the funding total calulation
```
Comment (zerolab) [inline opentech/public/utils/models.py]: Impliments -> Implements
calulation -> calculation

File: opentech/public/utils/models.py [hunk 2]
```diff
@@ -259,3 +260,39 @@ class Meta:
...
+
+    You still need to include the InlinePanel('funding') in the child class
```
Comment (zerolab) [inline opentech/public/utils/models.py]: Is there no way of making this so one just needs to do `panels + FundingMixin.panels` ?

File: opentech/public/people/models.py [hunk 3]
```diff
@@ -13,13 +13,12 @@
...
     InlinePanel,
     MultiFieldPanel,
-    PageChooserPanel,
     StreamFieldPanel
 )
 from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
 
 from opentech.public.utils.blocks import StoryBlock
-from opentech.public.utils.models import BasePage
+from opentech.public.utils.models import BasePage, BaseFunding, FundingMixin
```
Comment (zerolab) [inline opentech/public/people/models.py]: since this is shared, should utils be... on the same level of apply and public?

Comment (todd-dembrey) [reply opentech/public/people/models.py]: It is only shared in the public section of the site so didn't want it higher. equally it seems a bit weird in the utils folder.

File: opentech/public/utils/models.py (hunk 2 above)
Comment (todd-dembrey) [reply opentech/public/utils/models.py]: That is the correct pattern. Thank you.

File: opentech/public/utils/models.py [hunk 4]
```diff
@@ -259,3 +261,40 @@ class Meta:
...
+    class Meta(Orderable.Meta):
+        abstract = True
+
+
+class FundingMixin(models.Model):
+    '''Implements the funding total calculation
+
+    You still need to include the content panel in the child class
+    '''
+    content_panels = InlinePanel('funding', label="Funding"),
```
Comment (zerolab) [inline opentech/public/utils/models.py]: needs to be `[InlinePanel('funding', label="Funding")]`

File: opentech/public/projects/models.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    panels = [
+        PageChooserPanel('page', 'projects.ProjectPage'),
+    ]
+
+
+class ProjectFunding(BaseFunding):
+    page = ParentalKey('ProjectPage', related_name='funding')
+
+
+class ProjectPage(FundingMixin, BasePage):
```
Comment (zerolab) [inline opentech/public/projects/models.py]: This is missing the categories:
Core issues (select)
Objectives (select)
Beneficiaries (select)
Addressed problems (select)
Technology focus (select)
Regions (select)
Countries (select)

File: opentech/public/projects/widgets.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    def __init__(self, queryset, display):
+        self.queryset = queryset
+        self.display = display
+
+    def __iter__(self):
+        for choice in self.queryset.values_list(*self.display):
+            yield choice
+
+
+class LazyWidgets:
```
Comment (zerolab) [inline opentech/public/projects/widgets.py]: These will need some tests.

Comment (todd-dembrey) [reply opentech/public/projects/widgets.py]: Done.

File: opentech/public/projects/templates/projects/project_page.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                    {% endfor %}
+
+                {% endif %}
+            {% endwith %}
+
+        </div>
+    </div>
+    {% include "utils/includes/funding.html" %}
+
+    {% regroup page.category_options by category as categories %}
```
Comment (zerolab) [inline opentech/public/projects/templates/projects/project_page.html]: So, the widget shows the options in the correct order when editing the project 👍 
The order is not preserved here. 👎

File: opentech/public/projects/models.py [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+    content_panels = BasePage.content_panels + [
+        ImageChooserPanel('icon'),
+        FieldPanel('introduction'),
+        FieldPanel('status'),
+        StreamFieldPanel('body'),
+        InlinePanel('contact_details', label="Contact Details"),
+        InlinePanel('related_pages', label="Related Projects"),
+    ] + FundingMixin.content_panels + [
+        FieldPanel('categories', widget=CategoriesWidget),
```
Comment (zerolab) [inline opentech/public/projects/models.py]: Make this collapsible and possibly collapsed by default?

Other than this/ FE order, :shipit:

Comment (todd-dembrey) [reply opentech/public/projects/models.py]: image

Comment (todd-dembrey) [reply opentech/public/projects/models.py]: Looks a little weird with the duplicate name, but not terrible.

File: opentech/public/projects/templates/projects/project_page.html (hunk 7 above)
Comment (todd-dembrey) [reply opentech/public/projects/templates/projects/project_page.html]: `regroup` preserves order. I have updated the query to ensure the correct order is passed into the template

**Pull request merged**